IPython/core/compilerop.py

```diff
@@ -92,6 +92,9 @@ def __init__(self):
         # (otherwise we'd lose our tracebacks).
         linecache.checkcache = check_linecache_ipython
 
+        # Caching a dictionary { filename: execution_count } for nicely
+        # rendered tracebacks
+        self.filename_map = {}
```

Comment:
IIUC this isn't "execution count", it's the cell number. To me `execution_count` suggests, that this dictionary is used to remember how many times code with a given filename was run/compiled/etc. - it's a bit misleading.

Similarly, it's not _really_ the filename that is being remembered here as a key - it's a synthetic identifier that IPython gives its code blocks in the `code_name` func above. I think it would be worth it to try to try and signify it in the name of this variable, for posterity.

Maybe:

```suggestion
        # Caching dictionary { filename: cell_id } for nicely
        # rendered tracebacks
        self.co_filename_to_cell_id = {}
```

`co_filename` is the name of the attribute that the code objects use to keep this data.

Reply:
You could also get rid of this dictionary altogether, by creating a `code_name_to_cell_id` func (next to `code_name`), that would parse out the `number` out of the:

```
    return '<ipython-input-{0}-{1}>'.format(number, hash_digest[:12])
```

Then add a test for the `code_name_to_cell_id(code_name(code, number)) == number` assumption.
That way you could omit the `get_ipython` call, and therefore make it all a bit less coupled.